pages/lua_manual/api_reference.md

```diff
@@ -0,0 +1,1849 @@
+---
+layout: page
+title: Lua API Reference
+status: improve
+---
+
+All instantiable classes define a `new()` method which returns an object of that particular class. 
+
+With the exception of this `new()` method, all members listed will be member methods.
+
+## ElementDataGrid
+
+Inherits: `Element`{: .type }
+
+ElementDataGrid derives from Element. The data grid has the following methods and properties:
+
+### Properties
+
+| Types | Name |
+| ------------ | ---- |
+| `ElementDataGridRow`{: .type } | rows |
+
+
+### Methods
+
+| Return value | Name |
+| ------------ | ---- |
+| `nil`{: .type } | AddColumn(`string`{: .type } fields, `string`{: .type } formatter, `number`{: .type } initial_width, `string`{: .type } header_rml) |
```

Comment:
```suggestion
| `nil`{: .type } | <a href="#ElementDataGrid-AddColumn" class="lua_method">AddColumn(`string`{: .lua_type } fields, `string`{: .lua_type } formatter, `number`{: .lua_type } initial_width, `string`{: .lua_type } header_rml)</a> |
```
For linking to properties, and so we can style it later. (We could also write the `{:.lua_type}` with html tags if you prefer, I don't mind either way).

Reply:
Great, I am adding this. All table names (class names) are now links and the page links to them from the content table. Also as you did above, I am making all functions and properties to be linkable and their tables also link to them

Reply:
Great!